obspy/io/nordic/core.py

```diff
@@ -0,0 +1,1379 @@
+    """
+    if not hasattr(sfile, "readline"):
+        try:
+            with open(sfile, 'r') as f:
+                topline = f.readline()
+                if not len(topline.rstrip()) == 80:
+                    return False
+                f.seek(0)
+                for line in f:
+                    if line[79] in [' ', '1']:
+                        head_line = line
+                        break
+        except:
+            return False
+    else:
+        topline = sfile.readline()
+        if not len(topline.rstrip()) == 80:
+            return False
+        sfile.seek(0)
+        for line in sfile:
+            if line[79] in [' ', '1']:
+                head_line = line
+                break
+    if 'head_line' in locals():
+        try:
+            sfile_seconds = int(head_line[16:18])
+        except ValueError:
+            return False
+        if sfile_seconds == 60:
+            sfile_seconds = 0
+            add_seconds = 60
+        else:
+            add_seconds = 0
+        try:
+            UTCDateTime(int(head_line[1:5]), int(head_line[6:8]),
+                        int(head_line[8:10]), int(head_line[11:13]),
+                        int(head_line[13:15]), sfile_seconds,
+                        int(head_line[19:20]) * 100000) + add_seconds
+            return True
+        except:
+            return False
+    else:
+        return False
+
+
+def _int_conv(string):
+    """
+    Convenience tool to convert from string to integer.
+    If empty string return a 999 rather than an error.
+
+    >>> _int_conv('12')
+    12
+    >>> _int_conv('')
+    999
+    """
+    try:
+        intstring = int(string)
+    except:
+        intstring = 999
+    return intstring
+
+
+def _float_conv(string):
+    """
+    Convenience tool to convert from string to float.
+    If empty string return NaN rather than an error.
+
+    >>> _float_conv('12')
+    12.0
+    >>> _float_conv('')
+    999.0
+    >>> _float_conv('12.324')
+    12.324
+    """
+    try:
+        floatstring = float(string)
+    except:
+        # cjh Changed this to deal with QuakeML validation issues with 'NaN'
+        floatstring = float('999')
+    return floatstring
+
+
+def _str_conv(number, rounded=False):
+    """
+    Convenience tool to convert a number, either float or int into a string.
+    If the int is 999, or the float is NaN, returns empty string.
+
+    >>> _str_conv(12.3)
+    '12.3'
+    >>> _str_conv(12.34546, rounded=1)
+    '12.3'
+    """
+    string = str(number)
+    if (isinstance(number, float) and np.isnan(number)) or number == 999:
+        string = ' '
+    elif not rounded and \
+            (isinstance(number, float) or isinstance(number, int)):
+        if number < 100000:
+            string = str(number)
+        else:
+            exponant = int('{0:.2E}'.format(number).split('E+')[-1]) - 1
+            divisor = 10 ** exponant
+            string = '{0:.1f}'.format(number / divisor) + 'e' + str(exponant)
```

Comment:
Good point, I don't actually know if it requires that format for reading - I will have a play and see.  It definitely spits them out that way though.  From the manual that describes the format they say amplitudes are g7.1, it should cope with e.g. 1.1e6, but it looks like to get more precision you need e.g. 11.2e5 rather than 1.12e6?